### PR TITLE
Add SSE tests

### PR DESCRIPTION
**Pull request by davidpdrsn**
While reviewing https://github.com/tokio-rs/axum/pull/647 I thought some tests for SSE would help. So this adds at least some basic test coverage.

cc @SabrinaJewson

### CODE REVIEW

**Comment by davidpdrsn** [comment]
Actually it should be possible to write the keep alive test in a way that is less flaky and dependent on the timings of things. I'll play around.

**Comment by jplatte** [comment]
I know next to nothing about SSE, this should be reviewed by somebody else.

**Comment by davidpdrsn** [comment]
@SabrinaJewson are you up for reviewing this? You've done a bunch of stuff around SSE in axum.

File: axum/src/response/sse.rs [hunk 1]
```diff
@@ -451,11 +451,120 @@ impl KeepAliveStream {
...
+        let event_fields = parse_event(&res.chunk_text().await.unwrap());
+        assert_eq!(event_fields.get("event").unwrap(), "three");
+        assert_eq!(event_fields.get("retry").unwrap(), "30000");
+        assert_eq!(event_fields.get("id").unwrap(), "unique-id");
+        assert!(event_fields.get("comment").is_none());
+
+        assert!(res.chunk_text().await.is_none());
+    }
+
+    #[tokio::test]
```
Comment (SabrinaJewson) [inline axum/src/response/sse.rs:513]: If you change this to `#[tokio::test(start_paused = true)]` then we can use [`time::advance`](https://docs.rs/tokio/latest/tokio/time/fn.advance.html) to have the test run more deterministically and without any delay.

File: axum/src/response/sse.rs [hunk 2]
```diff
@@ -451,11 +451,120 @@ impl KeepAliveStream {
...
+        let event_fields = parse_event(&res.chunk_text().await.unwrap());
+        assert_eq!(event_fields.get("data").unwrap(), "one");
+
+        let event_fields = parse_event(&res.chunk_text().await.unwrap());
+        assert_eq!(event_fields.get("comment").unwrap(), "keep-alive-text");
+
+        let event_fields = parse_event(&res.chunk_text().await.unwrap());
+        assert_eq!(event_fields.get("data").unwrap(), "two");
+
+        assert!(res.chunk_text().await.is_none());
```
Comment (SabrinaJewson) [inline axum/src/response/sse.rs:547]: After this we can test that keep alives continue to be sent:

```rs
time::advance(KEEP_ALIVE_INTERVAL);
let event_fields = parse_event(&res.chunk_text().await.unwrap());
assert_eq!(event_fields.get("comment").unwrap(), "keep-alive-text");

assert!(res.chunk_text().await.is_none());
```

**Review by SabrinaJewson — COMMENTED** [review]
Looks good overall.

File: axum/src/response/sse.rs (hunk 1 above)
Comment (davidpdrsn) [reply axum/src/response/sse.rs]: Good idea! I've changed the keep alive tests!

**Comment by davidpdrsn** [comment]
@SabrinaJewson ready for another look!

**Review by SabrinaJewson — APPROVED** [review]
Seems fine 👍

**Pull request merged**